experimental/StandardFiniteFields/src/StandardFiniteFields.jl

```diff
@@ -0,0 +1,506 @@
+  vname = "x" * string(deg)
+  L, X = FiniteField(polynomial(K, lcoeffs), vname)
+  set_standard_finite_field!(L)
+  set_primitive_powers_in_tower_basis!(L, pmat)
+
+  return L
+end
+function _extension_with_tower_basis(
+  K::T,
+  deg::IntegerUnion,
+  lcoeffs::Vector,
+  b::FinFieldElem,
+) where {T<:FinField}
+  @assert parent(b) === K
+
+  dK = absolute_degree(K)
+  if dK == 1
+    then
+    println("_extension_with_tower_basis is running unoptimized...")
+  end
+  d = Int(dK * deg)
+  F = prime_field(K)
+  while length(lcoeffs) < deg
+    push!(lcoeffs, zero(K))
+  end
+  push!(lcoeffs, one(K))
+  pK = primitive_powers_in_tower_basis(K)
+
+  # The idea is to collect (bX)^i mod f for 1 in 0..d*dK-1
+  # and use this to compute the minimal polynomial of bX over F.
+  # Should we just form the polynomial and compute "mod"???
+  vec = zeros(F, d)
+  vec[1] = one(F)
+  v = zeros(K, Int(deg))
+  v[1] = one(K)
+
+  vecs = Vector{Vector{eltype(F)}}(undef, d)
+  pols = Vector{Vector{eltype(F)}}(undef, d)
+  pmat = zero_matrix(F, d, d)
+  poly = Vector{eltype(F)}[]
+
+  for i = 1:d+1
+    # println("i: ", i, " vec: ", vec, " v: ", v)
+    if i <= d
+      pmat[i, :] = vec
+    end
+
+    poly = zeros(F, i)
+    poly[end] = one(F)
+
+    w = copy(vec)
+    piv = findfirst(!iszero, w)
+    # TODO : figure out the purpose of this loop and FIX it
+    while piv != nothing && isassigned(vecs, piv)
+      x = -w[piv]
+      if isassigned(pols, piv)
+        # println("p: ", p, " piv ", piv, " pols[piv]: ", pols[piv])
+        poly[1:length(pols[piv])] += x .* pols[piv]
+      end
+      w[piv:d] += x .* @view vecs[piv][piv:d]
+      piv = findnext(!iszero, w, piv + 1)
+    end
+    # NOTE : exits the while loop when either piv == nothing, or vecs[piv] is undefined.
+    #        what happens if piv == nothing???
+    # println("Exiting loop with piv = ", piv)
+    if i <= d
+      # println(order(K), " ", i, " ", piv, " ", v, " ", w)
+      x = inv(w[piv])
+      poly .= x .* poly
+      w .= x .* w
+      pols[piv] = copy(poly)
+      vecs[piv] = copy(w)
+
+
+      # Multiply by vX and find the next vec in the Tower Basis
+      v = collect(
+        coefficients(
+          mod(polynomial(K, pushfirst!(b .* v, zero(K))), polynomial(K, lcoeffs)),
+        ),
+      )
+
+      while length(v) < deg
+        push!(v, zero(K))
+      end
+      # println("new v:", v)
+      vec = vcat(map(a -> coords(a) * pK, v)...)
+
+    end
+  end
+  # Now p is the minimal polynomial over F
+  # pmat gives the primitive powers in the tower basis for the new extension
+
+  vname = "x" * string(d)
+  L, X = FiniteField(polynomial(F, poly), vname)
+  set_standard_finite_field!(L)
+  set_primitive_powers_in_tower_basis!(L, pmat)
+
+  return L
+
+end
+
+
+@doc raw"""
+    standard_finite_field(p::Union{ZZRingElem, Integer}, n::Union{ZZRingElem, Integer}) -> FinField
+
+Returns a finite field of order $p^n$.
+
+# Examples
+```jldoctest
+julia> standard_finite_field(3, 24)
```

Comment:
This can't work because `standard_finite_field` is not exported by `Oscar`

Reply:
ah, okay. So should I comment out the example and then the doctest will pass? Or can it be tested as `Oscar.standard_finite_field`?

Reply:
This should be fine now due to the export outside of your module in line 509.

Reply:
This should be fine now due to the export outside of your module in line 509.